schema_based_on_diagram.sql

```diff
@@ -0,0 +1,54 @@
+CREATE DATABASE clinic;
+
+CREATE TABLE patients (
+  id SERIAL NOT NULL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  date_of_birth DATE
+);
+
+CREATE TABLE medical_histories (
+  id SERIAL NOT NULL PRIMARY KEY,
+  admitted_at TIMESTAMP NOT NULL,
+  patient_id INT NOT NULL,
+  status VARCHAR(255) NOT NULL,
+  FOREIGN KEY (patient_id) REFERENCES patients (id)
+);
+
+CREATE TABLE treatments (
+  id SERIAL NOT NULL PRIMARY KEY,
+  type VARCHAR(255) NOT NULL,
+  name VARCHAR(255) NOT NULL
+);
+
+CREATE TABLE invoices (
+  id SERIAL NOT NULL PRIMARY KEY,
+  total_amount DECIMAL NOT NULL,
+  generated_at TIMESTAMP NOT NULL,
+  payed_at TIMESTAMP,
+  medical_history_id INT NOT NULL,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+);
+
+CREATE TABLE invoice_items (
+  id SERIAL NOT NULL PRIMARY KEY,
+  unit_price DECIMAL NOT NULL,
+  quantity INT NOT NULL,
+  total_price DECIMAL NOT NULL,
+  invoice_id INT NOT NULL,
+  FOREIGN KEY (invoice_id) REFERENCES invoices (id),
+  treatment_id INT NOT NULL,
+  FOREIGN KEY (treatment_id) REFERENCES treatments (id)
+);
+
+-- To create the many to many relationship between medical_histories and treatments
+CREATE TABLE histories_treatments (
+  history_id INT REFERENCES medical_histories(id) ON DELETE CASCADE,
+  treatment_id INT REFERENCES treatments(id) ON DELETE CASCADE
+);
+
+CREATE INDEX patients_id ON patients(id);
```

Comment:
- [ ] according to the diagram, please ensure that the indexes are created for the foreign keys. the "join table" for the many-to-many relastionships is ok.  
eg 
* `CREATE INDEX ON medical_histories (patient_id);`
* `CREATE INDEX ON invoices (medical_history_id);`
* add for the invoice items. 
nice work